### ISSUE TITLE
nixos-rebuild test (and switch) causes pam_mount to remount

### ISSUE THREAD

**Issue by peterhoeg** [NixOS/nixpkgs#44586]
## Issue description

Within the last couple of weeks, doing ```nixos-rebuild test``` causes pam_mount to remount encrypted home directories and thus asking for passwords. This wasn't the case in the past.

Relevant output:
```
building '/nix/store/p6bflh7n5zy2dql8l45mix9qnzq65hbk-nixos-system-mildred-18.09.git.98592c5da79M.drv'...
activating the configuration...
setting up /etc...
reenter password for pam_mount:
(mount.c:68): Messages from underlying mount program:
(mount.c:72): crypt_activate_by_passphrase: File exists
(pam_mount.c:522): mount of /dev/mapper/vg0-lv_home_peter failed
kbuildsycoca5 running...
```

### Steps to reproduce

1. Set ```users.users.some_user.cryptHomeLuks = "/dev/mapper/some_lv";```
2. Execute ```nixos-rebuild test```

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.17.11, NixOS, 18.09.git.98592c5da79M (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.0.4`
 - channels(root): `"nixos-18.03pre121670.4cbbecc85db"`
 - channels(peter): `""`
 - nixpkgs: `/home/peter/src/active/nixpkgs_unstable`

**Comment by peterhoeg** [comment]
Duplicate of #47577.

**Comment by peterhoeg** [comment]
This isn't actually a duplicate. With the fix in #47842, ```nixos-rebuild switch``` still prompts once (versus twice before) so must be another ```system.activationScript``` that causes this. This isn't critical though, just annoying.

**Comment by peterhoeg** [comment]
I've just patched it out locally, but this needs to be fixed properly.

**Comment by peterhoeg** [comment]
keep

**Comment by Aleksanaa** [comment]
This issue reappleared in nixos unstable and it's the same as you described. And home manager is not working for me, possibly because related problems. Any ideas? Or should I open a new issue?

**Comment by peterhoeg** [comment]
So I have been carrying the mentioned patch on my side since this was first mentioned. I have a laptop running unstable with an encrypted home and so far there has been zero problems. Maybe you can share your nixos config?

**Comment by Aleksanaa** [comment]
> So I have been carrying the mentioned patch on my side since this was first mentioned. I have a laptop running unstable with an encrypted home and so far there has been zero problems. Maybe you can share your nixos config?

Do you still have these error indicating pam_mount tries to mount again (not remount):

```
(mount.c:68): Messages from underlying mount program:
(mount.c:72): Device _dev_nvme0n1p3 already exists.
(mount.c:72): crypt_activate_by_passphrase: File exists
(mount.c:72): Device _dev_nvme0n1p3 is still in use.
```
When running `nixos-rebuild switch`?

**Comment by peterhoeg** [comment]
Yep, but it doesn't matter. The patch simply makes it not prompt for a password when the activation scripts tries to set up the mount, but as it's already there, you get the messages mentioned above, which is ok.
